Edit-message

taniokatomoya opened the pull request:
#what
incremental_search.jsの作成
#why
グループ画面でのユーザー検索をインクリメンタルサーチで検索できるようにするため

Commit:
Edit-message

#what
jqueryとjbuilderの作成
#why
メッセージの非同期通信

ta910 commented:
コミットは複数に分けるようにしましょう。

ta910 commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,54 @@
+$(function(){
+  function buildHTML(message){
+    var html = `<div class="message">
```
もっとhtmlライクにインデントは調節しましょう

ta910 commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,54 @@
+    })
+})
+});
+
```
不要な空行やコメントは消しておきましょう

ta910 commented on app/assets/stylesheets/application.scss:
```diff
@@ -7,4 +7,11 @@
+@import "config/colors"; /*色変数*/
+@import "modules/flash";
+@import "modules/group";
+// @import "modules/chat";
```
こちらも

ta910 commented on app/controllers/groups_controller.rb:
```diff
@@ -0,0 +1,42 @@
+    if @group.save
+      redirect_to root_path, notice: 'グループが作成されました'
+    else
+      render :new
```
エラーメッセージ出しましょう

ta910 commented on app/controllers/groups_controller.rb:
```diff
@@ -0,0 +1,42 @@
+    if @group.update(group_params)
+      redirect_to group_messages_path(@group), notice: 'グループを編集しました'
+    else
+      render :edit
```
こちらも

ta910 commented on app/controllers/groups_controller.rb:
```diff
@@ -0,0 +1,42 @@
+  private
+
+  def group_params
+    params.require(:group).permit(:name, { :user_ids => [] })
```
ハッシュロケットは古い記法なので避けましょう

ta910 commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,19 @@
+class UsersController < ApplicationController
```
user関連の動作は全てdeviseで行なってくれるのでusers_controllerはいらないです

ta910 commented:
画像はgithubに上げないようにしましょう

ta910 commented:
一度に見るファイルが多すぎるのでPRはもっとこまめに出しましょう

Commit:
Chatspaceメッセージの非同期通信

#what
メッセージの非同期通信
#why
jqueryとjbuilderの作成

Commit:
Merge branch 'master' into edit-message

Commit:
WHAT
incremental_search.jsの作成
WHY
インクリメンタルサーチ実装のため

taketnk commented on app/views/messages/_left.content.haml:
```diff
@@ -1,12 +1,23 @@
-      %i.fa.fa-pencil-square-o
-    %a{href: '#'}
-      %i.fa.fa-cog
+    = current_user.name
```
こちらのファイル名と拡張子おかしくないですか？

taketnk commented on app/views/messages/_maincontent__bottom.haml:
```diff
@@ -1,6 +1,4 @@
-  %p.maincontent__bottom--username username
-  %p.maincontent__bottom--dateandtime 2018/11/11 11:11:11
-  %p.maincontent__bottom--message メッセージ1
+  = render partial: 'message', collection: @messages
```
こちらも、html.hamlだと思います！

taketnk commented on app/views/messages/_maincontent__toukou.haml:
```diff
@@ -1,14 +1,13 @@
-  <label for="send" class="filesend">Send<input type="submit" name="botan" value="send">
-  </label>
-  </form>
+  .form
```
こちらもhtml.hamlで

taketnk commented:
gitignoreに画像をgitの追跡対象にしない記述はされましたでしょうか？

taketnk commented:
せめてwhatとwhyには、こちらのプルリクエストが何をしているのか詳細に書いていただけると見る側として助かります
🙏

Commit:
edit gitignore

edit gitignore

ta910 commented:
ファイルが多すぎることで修正したファイルがどこなのか、どこを新しく実装したかがわかりません。毎回レビュー時に全ファイルを見るのは負担が大きいのでPRは分けるようにしましょう。
せめて、PRにコメントでどの部分、どのファイルをレビューしてもらいたいのか書いていただきたいです。

taniokatomoya commented:
app/assets/javascripts/incremental_search.js
のレビューをお願いします。よろしくお願いします。

Commit:
Edit incremental_search

What
incremental_search.jsの作成
why
グループ画面でのユーザー検索をインクリメンタルサーチで検索できるようにするため
app/assets/javascripts/incremental_search.jsのレビューをお願いします

taniokatomoya commented:
app/assets/javascripts/incremental_search.jsのレビューをお願いします。よろしくお願いします。

ta910 commented on app/assets/javascripts/incremental_search.js:
```diff
@@ -0,0 +1,68 @@
+          appendUser(user);
+        });
+      }else{
+        appendNoUser("一致する人がいません");
```
ここで"一致する人がいません"を引数に取っていますが、appendNoUser内でこの引数を用いていないのでいらないのではないでしょうか？

Commit:
Edit incremental_search.js

What
incremental_search.jsの編集
Why
インクリメンタルサーチ実装のため

Commit:
edit

What
incremental_search.jsの編集
Why
グループ画面でのユーザー検索をインクリメンタルサーチで検索できるようにするため
